### PR TITLE
Better tests for `IsolatedEnvironment` and some minor fixes

### PR DESCRIPTION
**Pull request by FFY00**
This tests if the required environment directories are provided by
IsolatedEnvironment.

Signed-off-by: Filipe Laíns <lains@archlinux.org>

### CODE REVIEW

File: src/build/env.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def _symlink_relative(self, path):  # type: (Optional[str]) -> None
         if not path:  # pragma: no cover
             return
 
-        prefix = sysconfig.get_config_var('prefix')
+        prefix = sysconfig.get_config_var('base' if sys.version_info[0] == 2 else 'prefix')
```
Comment (gaborbernat) [inline src/build/env.py:76]: Could be worth documenting why the discrepency?

File: src/build/env.py [hunk 2]
```diff
@@ -130,7 +130,7 @@ def __enter__(self):  # type: () -> IsolatedEnvironment
 
         self._symlink_relative(sysconfig.get_path('include'))
         self._symlink_relative(sysconfig.get_path('platinclude'))
-        self._symlink_relative(sysconfig.get_config_var('srcdir'))
+        self._symlink_relative(sysconfig.get_config_var('LIBPL'))
```
Comment (gaborbernat) [inline src/build/env.py:133]: also we could document why this is needed 🤷

Comment (FFY00) [reply src/build/env.py:133]: Commit description

> LIBPL contains the files necessary for embedding Python in C, I wrongly
assumed srcdir was used for this. srcdir will point to LIBPL in Python
3, but not in Python2.

File: src/build/env.py (hunk 1 above)
Comment (FFY00) [reply src/build/env.py]: Oh, it was leftover from my initial tries. I just checked and this is actually not needed. Thanks :blush:

File: src/build/env.py (hunk 2 above)
Comment (gaborbernat) [reply src/build/env.py:133]: Ah, I see. Often I make it a code comment, as the commit history gets lost easier.

**Pull request merged**